main/provider/helpers.ts

```diff
@@ -68,6 +70,13 @@ export function feeTotalOverMax (rawTx: TransactionData, maxTotalFee: number) {
 
 export function getRawTx (newTx: RPC.SendTransaction.TxParams, accountId: string | undefined): TransactionData {
   const { gas, gasLimit, gasPrice, data, value, type, to, ...rawTx } = newTx
+  const getNonce = () => {
+    if (isHexString(rawTx.nonce)) {
+      return rawTx.nonce
+    }
+    const nonceBN = new BN(rawTx.nonce as string)
+    return nonceBN.isNaN() ? undefined : addHexPrefix(nonceBN.abs().integerValue().toString(16))
```

Comment:
think this line can be simplified to `return nonceBN.isNaN() ? undefined : addHexPrefix(nonceBN.toString(16))`

Reply:
I guess it depends on how many edge cases / invalid nonce scenarios we want to be covering...

Reply:
yeah maybe we can explicitly check for negatives, but either way we should just reject it and not convert it to positive